### PR TITLE
Fixes for recent changes

### PR DESCRIPTION
**Pull request by yut23**
This has several fixes/improvements that I've been using for a while, which I've pulled out into separate commits:

* try to avoid a race condition that would sometimes result in missing or duplicated URLs
* fix the broken formatting after Google added site icons and put the URL on a separate line
* add support for dark mode (use a lighter shade of green for the URL if dark mode is enabled)

### CODE REVIEW

File: userscript.js [hunk 1]
```diff
@@ -70,9 +74,9 @@
     var bettered = false;
 
     var runBetterGoogle = function() {
-        if (prevResultCount != document.querySelectorAll('.g .yuRUbf').length) {
+        if (prevResultCount != document.querySelectorAll('.g .yuRUbf .rnBE4e').length) {
             document.querySelectorAll('.g .yuRUbf').forEach(betterGoogleRow);
-            prevResultCount = document.querySelectorAll('.g .yuRUbf').length;
+            prevResultCount = document.querySelectorAll('.g .yuRUbf .rnBE4e').length;
```
Comment (yut23) [inline userscript.js:79]: I'm now getting a different version of the "More options" button that has the `csDOgf` class and does some weird stuff with absolute positioning. I'm going to try without this hunk locally and see if things still work fine.

**Comment by drwonky** [comment]
You will probably want to rebase this since a fix was just released.  Please attach some screenshots of what these changes do so I can better understand.

**Comment by yut23** [comment]
Screenshots in light mode and dark mode:

| Before | After |
|--------|--------|
| light, master | light, pr 19 |
| dark, master | dark, pr 19 |

**Review by drwonky — APPROVED** [review]
The results look good, thanks!

**Comment by drwonky** [comment]
When you are ready, merge this PR, I approved it.

**Review by drwonky — APPROVED** [review]

**Comment by yut23** [comment]
I don't have write access to this repo, so you'll have to merge it.

**Comment by drwonky** [comment]
> I don't have write access to this repo, so you'll have to merge it.

If the PR is approved, I assumed you could merge it?  That's how it works with CODEOWNERS in GHE at work.  As long as someone with write access approves the PR, anyone can hit the merge button, then the ownership record is preserved for commits.

**Pull request merged**